tests/integration/integration_test_case.py

```diff
@@ -22,78 +54,163 @@ def setUp(self):
             password = os.getenv(password_name, dev_default)
             vars(settings)[password_name] = password  # assigns attribute to this module
 
-        self.application = create_app()
-        self.client = self.application.test_client()
+    def _setUpApp(self):
+        self._application = create_app()
+        self._client = self._application.test_client()
+        # Clear any existing state before the test starts
+        self.clearDatabase()
```

Comment:
yeah tricky as the underlying base class (`TestCase`) uses that format but PEP8 is `_` based; so I think the `Assert` calls should be consistent with TestCase but not sure about these ones; as `setUp` is from `TestCase` but the private functions could be `_` based?